src/definitions/modules/modal.js

```diff
@@ -8,18 +8,19 @@
  *
  */
 
+// eslint-disable-next-line no-shadow-restricted-names
 (function ($, window, document, undefined) {
     'use strict';
 
     function isFunction(obj) {
         return typeof obj === 'function' && typeof obj.nodeType !== 'number';
     }
 
-    window = (typeof window != 'undefined' && window.Math == Math)
+    window = window !== undefined && window.Math === Math
         ? window
         : globalThis;
 
-    $.modal = $.fn.modal = function (parameters) {
```

Comment:
Dont remove the short cut call without an attached element. It was explcitely added to 2.9.0